libs/island-ui/core/src/lib/InputImageUpload/InputImageUpload.tsx

```diff
@@ -0,0 +1,340 @@
+import React, { useMemo } from 'react'
```

Comment:
This component should probably not live inside core. It's making a GraphQL query which would break every application that is not coupled by the shared GraphQL API.
Should most likely be moved to `shared/components`.

Reply:
I'd also argue that since it has application-system specific logic it should be in libs/application/ui-components or something.

Would be nice though (now or later) to split up the component logic so the same UI can be used with other "backends" later.

Reply:
Yeah, I agree on libs/application/ui-components ... just not core 😄 .

But I agree, the InputImageUpload could/should be more generic so it can be moved to `core`, and then have another wrapper component in libs/application/ui-components that uses InputImageUpload component from core alongside queries and application specific logic.

Reply:
since this is a part of the buildFileUploadField and is sort of an addon to it (used if the template field has 'forImageUpload' prop), im not sure i understand why it needs to be moved away from InputFileUpload

but i will do so if u say so! :D

Reply:
aaaah okok, yes, due to the query within this file?

Reply:
Yeah. Core components shouldn't have any side effects.